extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -457,6 +456,31 @@ export class SubscriptionService {
       platform.decorateUI(element, action, options);
     });
   }
+
+  /**
+   * Evaluates platforms and select the one to be selected for login.
+   * @return {!Promise<!./subscription-platform.SubscriptionPlatform>}
+   */
+  scoreBasedLogin() {
+    return this.platformStore_.getAllPlatforms().then(platforms => {
```

Comment:
We are taking some risk here getting into popup blocker situation. What if we went synchronous here with strictly already loaded platforms? Local is always there, so we always have one option. In other words, maybe `getAvailablePlatforms()` method instead? Otherwise "login" is only enhanced - there's no significant change in behavior. If we fail to enhance something - it's still functional.